Framework/API/src/AnalysisDataService.cpp

```diff
@@ -362,13 +376,24 @@ void AnalysisDataServiceImpl::setIllegalCharacterList(
 /**
  * Checks the name is valid
  * @param name A string containing the name to check. If the name is invalid a
- * std::invalid_argument is thrown
+ * std::invalid_argument or std::runtime_error is thrown
+ * @param group A WorkspaceGroup shared ptr which will be evaluated if it is
+ * valid then it will check it's container for the same name that is passed if
+ * true throws std::runtime_error else nothing.
  */
-void AnalysisDataServiceImpl::verifyName(const std::string &name) {
+
+void AnalysisDataServiceImpl::verifyName(
+    const std::string &name,
+    const boost::shared_ptr<API::WorkspaceGroup> &group) {
   const std::string error = isValid(name);
   if (!error.empty()) {
     throw std::invalid_argument(error);
   }
+
+  if (group && group->contains(name)) {
+    throw std::runtime_error(
```

Comment:
I think this should be an invalid_argument error (both checks are valid name checks so should throw the same thing)